components/applicationTracker/ApplicationTracker.js

```diff
@@ -21,21 +27,145 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
-export default function ApplicationTracker() {
+const APPLICATION_STATUS_TYPES = [
+  {
+    value: 'submitted',
+    label: 'Application Submitted',
+  },
+  {
+    value: 'phone-screen',
+    label: 'Phone Screen',
+  },
+  {
+    value: 'interview',
+    label: 'Interviews',
+  },
+  {
+    value: 'offer-received',
+    label: 'Offer Received',
+  },
+  {
+    value: 'in-negation',
+    label: 'In Negation',
+  },
+  {
+    value: 'offer-accepted',
+    label: 'Offer Accepted',
+  },
+  {
+    value: 'offer-rejected',
+    label: 'Offer Rejected',
+  },
+];
+
+export async function logApplication(userDocRef, applicationDetails) {
+  const data = {
+    config: {
+      applicationStatusTypes: APPLICATION_STATUS_TYPES,
+    },
+    timestamp: firebase.firestore.FieldValue.serverTimestamp(),
+    ...applicationDetails,
+  };
+
+  return userDocRef.collection('applicationLogEntries').add(data);
+}
+
+export default function ApplicationTracker({ allApplicationLogEntries }) {
   const classes = useStyles();
+  const { userDocRef } = useAuth();
+  const applications = allApplicationLogEntries.map(item => ({
+    id: item.id,
+    document: item.data(),
+  }));
+
+  const handleSave = async () => {
+    const statusEntry = {
+      id: 1, // starting with index 1 when new application added
```

Comment:
So if I am adding an application, I am setting the `id` on the application's statusEntry to 1? (do we maybe want to have a more descriptive name, like statusId?)

Reply:
yes, when adding an application it's always the first status. I agree we could use a more descriptive name.

Reply:
statusId is good